File: examples/governance/compliance_domain/lib/compliance_domain/exemption/events/revoked_exemption.rb
module ComplianceDomain
  class Exemption
    module Events
      class RevokedExemption
        attr_reader :aggregate_id, :exemption_id, :model_id, :policy_id, :requirement, :reason, :approved_by_id, :approved_at, :expires_at, :scope, :status, :occurred_at

        def initialize(aggregate_id: nil, exemption_id: nil, model_id: nil, policy_id: nil, requirement: nil, reason: nil, approved_by_id: nil, approved_at: nil, expires_at: nil, scope: nil, status: nil)
          @aggregate_id = aggregate_id
          @exemption_id = exemption_id
          @model_id = model_id
          @policy_id = policy_id
          @requirement = requirement
          @reason = reason
          @approved_by_id = approved_by_id
          @approved_at = approved_at
          @expires_at = expires_at
          @scope = scope
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/revoked_exemption.go
package domain

import "time"

type RevokedExemption struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	ExemptionId string `json:"exemption_id"`
	ModelId string `json:"model_id"`
	PolicyId string `json:"policy_id"`
	Requirement string `json:"requirement"`
	Reason string `json:"reason"`
	ApprovedById string `json:"approved_by_id"`
	ApprovedAt time.Time `json:"approved_at"`
	ExpiresAt time.Time `json:"expires_at"`
	Scope string `json:"scope"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e RevokedExemption) EventName() string { return "RevokedExemption" }

func (e RevokedExemption) GetOccurredAt() time.Time { return e.OccurredAt }
